config_types.cpp

```diff
@@ -263,3 +266,93 @@ const char* server_addr::get_last_error(void) const
 {
     return gai_strerror(m_last_error);
 }
+
+config_cpu_list::config_cpu_list(unsigned int cores)
+{
+    for (unsigned int i = 0; i < cores; ++i)
+        cpu_list.push_back(i);
+
+    next_cpu = cpu_list.begin();
+}
+
+config_cpu_list::config_cpu_list(std::string str)
+{
+    std::stringstream ss(str);
+
+    if (str.find(' ') != std::string::npos)
+        return;
+
+    char separator = str[str.find_first_of(",-")];
+
+    int i;
+    while (ss >> i)
+    {
+        cpu_list.push_back(i);
+
+        if (ss.peek() == separator)
+            ss.ignore();
+        else if (ss.eof())
+            break;
+        else {
+            cpu_list.clear();
+            break;
+        }
+    }
+
+    next_cpu = cpu_list.begin();
+}
+
+config_cpu_list::config_cpu_list(const config_cpu_list& copy) :
+        cpu_list(copy.cpu_list)
+{
+    next_cpu = cpu_list.begin();
+}
+
+config_cpu_list& config_cpu_list::operator=(const config_cpu_list& cl)
+{
+    if (this == &cl)
+        return *this;
+
+    cpu_list = cl.cpu_list;
+    next_cpu = cpu_list.begin();
+    return *this;
+}
+
+bool config_cpu_list::is_defined() const
+{
+    return cpu_list.size() > 0;
+}
+
+bool config_cpu_list::is_legal() const
```

Comment:
don't you want to validate the cpu list in the constructor?

Reply:
btw, what do you in case an invalid core was provided? do you ignore it and continue?

Reply:
the changed parser checks the validity of the list provided and returns an error when the input is invalid, this is compatible with the rest of the code